daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/parsers/Parser.scala

```diff
@@ -173,31 +141,38 @@ class SeqCompParser(context: RuntimeData, val childParsers: Array[Parser])
 
 }
 
-class AltCompParser(context: RuntimeData, val childParsers: Seq[Parser])
-  extends ParserObject(context) {
+class AltCompParser(ctxt: RuntimeData, val childParsers: Seq[Parser])
+  extends ParserObject(ctxt) {
 
   override lazy val childProcessors = childParsers
 
   override def nom = "alt"
 
   def parse(pstate: PState): Unit = {
     var pBefore: PState.Mark = null
-
-    pstate.pushDiscriminator
-    var diagnostics: Seq[Diagnostic] = Nil
-    var i = 0
-    var parser: Parser = null
-    val limit = childParsers.length
-    var returnFlag = false
-    while (!returnFlag && i < limit) {
-      parser = childParsers(i)
-      i += 1
-      log(LogLevel.Debug, "Trying choice alternative: %s", parser)
-      pBefore = pstate.mark("AltCompParser1")
-      try {
-        parser.parse1(pstate)
-      } catch {
-        case d: SchemaDefinitionDiagnosticBase => {
+    try {
```

Comment:
Ok, so this code got colbberred. There's no catch for this try.... does't even compile!